src/run/lib/Main.ml

```diff
@@ -10,6 +10,7 @@ let safe_run f =
   try f ()
   with e ->
     let trace = Printexc.get_backtrace () in
+    flush stdout;
```

Comment:
I never really understood when we need to flush ... I usually assume if your
printed string has an \n then this will flush it. I am probably wrong.

Reply:
Here I make sure that anything to be printed on stdout gets printed before we print something on stderr (or exit with nonzero status, causing the output to be lost) to preserve chronological order.

OCaml's predefined out_channels stdout and stderr don't flush automatically when calling a printf function, in contrast to libc streams (where printf to stdout flushes with each newline, and printf to stderr flushes for each call).

Some ocaml functions do flush stdout: print_newline, print_endline, some read_* functions. Those are tricky to remember.

Explicit flushing with printf functions can be done with `%!` e.g. `Printf.printf "Hello!\n%!"`.